src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.Unix.cs

```diff
@@ -39,7 +37,17 @@ public UnixDomainSocketEndPoint(string path)
             }
 
             _path = path;
-            _encodedPath = s_pathEncoding.GetBytes(_path);
+
+            // Pathname socket addresses should be null-terminated.
+            // Linux abstract socket addresses start with a zero byte, they must not be null-terminated.
+            bool isAbstract = path.Length > 0 && path[0] == '\0';
+            int length = s_pathEncoding.GetByteCount(path);
+            if (!isAbstract)
+            {
+                length++;
+            }
+            _encodedPath = new byte[length];
+            s_pathEncoding.GetBytes(path, 0, path.Length, _encodedPath, 0);
```

Comment:
It'd be good to assert that the length we get back matches what we expected to the encoding to produce.

Reply:
Debug.Assert?

Reply:
Yup (but obviously as a separate statement so this required call doesn't evaporate in release builds :))